Sources/Credentials/CredentialsTokenTLL.swift

```diff
@@ -0,0 +1,57 @@
+/**
+* Copyright IBM Corporation 2016, 2017
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+**/
+
+import Kitura
+import KituraNet
+import Foundation
+
+/// Protocol to make it easier to add token TLL to credentials plugins.
+public protocol CredentialsTokenTLL {
```

Comment:
I successfully signed the CLA. Thanks and thanks for your comments. Will look at these closely shortly. (Definite oops on TLL! :) ).

Reply:
See what you think. I pushed your concept a little further. Here's an example of an authenticate method:

```
    public func authenticate(request: RouterRequest, response: RouterResponse,
                             options: [String:Any], onSuccess: @escaping (UserProfile) -> Void,
                             onFailure: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
                             onPass: @escaping (HTTPStatusCode?, [String:String]?) -> Void,
                             inProgress: @escaping () -> Void) {
        
        guard let type = request.headers["X-token-type"], type == name else {
            onPass(nil, nil)
            return
        }
        
        guard let token = request.headers["access_token"],
            let accountId = request.headers["X-account-id"] else {
            onFailure(nil, nil)
            return
        }
            
        self.accountId = accountId
        getProfileAndCacheIfNeeded(token: token, options: options, onSuccess: onSuccess, onFailure: onFailure)
    }
```

Reply:
By the way-- I didn't use a closure directly in the `getProfileAndCacheIfNeeded` method as the syntax got messy. Mostly due to the fact that the closure itself needed to be async because it will typically make networking calls.